source/daplink/interface/target_flash.c

```diff
@@ -253,15 +307,38 @@ static error_t target_flash_erase_chip(void)
 {
     if (g_board_info.target_cfg){
         error_t status = ERROR_SUCCESS;
-        const program_target_t *const flash = g_board_info.target_cfg->flash_algo;
-
-        status = flash_func_start(FLASH_FUNC_ERASE);
-        if (status != ERROR_SUCCESS) {
-            return status;
-        }
-        if (0 == swd_flash_syscall_exec(&flash->sys_call_s, flash->erase_chip, 0, 0, 0, 0)) {
-            return ERROR_ERASE_ALL;
-        }
+        program_target_t * flash;
+        flash_region_info_t * flash_region = g_board_info.target_cfg->extra_flash;
+        
+        do {
+            if (current_flash_algo != g_board_info.target_cfg->flash_algo){
```

Comment:
Since this is in your **do** loop, it will switch to the main flash algo every time you select an extra region algo using the **while** on line 330.

Reply:
Oh yeah totally misplaced this! Thanks